app/models/user/sign_in.rb

```diff
@@ -0,0 +1,47 @@
+# == Schema Information
+# Schema version: 20220225214524
+#
+# Table name: user_sign_ins
+#
+#  id         :bigint           not null, primary key
+#  user_id    :bigint
+#  ip         :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  country    :string
+#
+# Record medadata about User sign in activity.
+class User::SignIn < ApplicationRecord
+  default_scope { order(created_at: :desc) }
+
+  belongs_to :user, inverse_of: :sign_ins
+
+  before_create :create?
+
+  def self.purge
+    where('created_at < ?', retention_days.days.ago).destroy_all
```

Comment:
```suggestion
    where(created_at: (...retention_days.days.ago)).destroy_all
```

Reply:
To me the former is much clearer. Just resolving for now but happy for a PR later to change.